src/fu-engine.c

```diff
@@ -1407,6 +1407,12 @@ fu_engine_install_blob (FuEngine *self,
 	if (plugin == NULL)
 		return FALSE;
 
+	/* in case another device caused us to go into replug before starting */
+	if (!fu_device_list_wait_for_replug (self->device_list, device, error)) {
+		g_prefix_error (error, "failed to wait for pre-update replug: ");
+		return FALSE;
+	}
```

Comment:
do we need to re-get the device again here afterwards? (like other calls to fu_device_list_wait_for_replug?)

Reply:
No I don't believe so.
It comes back as the same ID, and in `fu_device_list_replace` this code is called which should handle setting it back up properly.
```
	g_set_object (&item->device_old, item->device);
	g_set_object (&item->device, device);
	fu_device_list_emit_device_changed (self, device);
```

I'd actually question if the extra calls are even needed in the other instances of `fu_engine_install_blob` given that behavior.

Reply:
I did some experiments and tearing that out in the other places replug uses leads to failures.  So I've added the re-get the device afterwards code here (and made sure it still works).